x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_error_grouping.yml

```
@@ -39,18 +39,23 @@ setup:
           - create: {}
           - '{"@timestamp": "2017-06-22", "error": {"log": {"message": ""}, "exception": [{"message": "exception_used"}]}}'
 
+          # Non-empty error.exception.message used from array
+          - create: {}
+          - '{"@timestamp": "2017-06-22", "error": {"log": {"message": ""}, "exception": [{"message": "first_exception_used"}, {"message": "2_ignored"}]}}'
```

```suggestion
          - '{"@timestamp": "2017-06-22", "error": {"log": {"message": ""}, "exception": [{"message": ["first_exception_used", "2_ignored"]}]}}'
```
Not sure if I am reading it wrong but it seems that we are adding the code for a message that could be a slice.

---

When using synthetic source, the source gets synthesized on-the-fly when accessing `params['_source']` in the runtime field script. The synthesized `_source` looks a bit differently compared to the original one: `{"error":{"exception":"message":["first_exception_used"," 2_ignored"]}}`.

---

Thanks for the explanation!